dstyle.dd

```diff
@@ -395,6 +395,38 @@ class MyClass
 -------------------------------
 )
 
+$(LISTSECTION phobos_documentation, Documentation,
+    $(LI Every public symbol be exposed on the documentation:)
+---
+/// A public symbol
+enum myFancyConstant;
+---
+    $(LI Every public function should have a Ddoc description and documented
+    `Params:` and `Returns:` sections (if applicable):)
+---
+/**
+Checks whether a number is positive.
+`0` isn't considered as positive number.
+
+Params:
+    number = number to be checked
+
+Returns: `true` if the number is positive, `0` otherwise.
+
+See_Also: $(LREF isNegative)
+*/
+bool isPositive(int number)
+{
+    return number > 0;
+}
+---
+    $(LI Text in sections (e.g. `Params:`, `Returns:`, `See_Also`) should be indented by one level if spans more than the line of the section.)
+    $(LI Documentation comments should not use more than two stars `/**` in the header line.)
+    $(LI Block comments (`/**`) should be used - not nesting block comments (`/++`))
+    $(LI Global functions shouldn't indent their documentation block nor use stars as indentation.)
```

Comment:
```
/**
Capitalize the first character of $(D s) and convert the rest of $(D s) to
lowercase.

Params:
    input = The string to _capitalize.

Returns:
    The capitalized string.

See_Also:
     $(REF asCapitalized, std,uni) for a lazy range version that doesn't allocate memory
*/
```

vs.

```
/**
 * Capitalize the first character of $(D s) and convert the rest of $(D s) to
 * lowercase.
 *
 * Params:
 *     input = The string to _capitalize.
 *
 * Returns:
 *     The capitalized string.
 *
 * See_Also:
 *      $(REF asCapitalized, std,uni) for a lazy range version that doesn't allocate memory
 */
```

While I can understand that the latter is used for symbols in structs, classes etc., I generally don't like the extra work and noise added by the stars.

Reply:
If we're going to require `/++ +/` or `/** */`, can we _please_ require `/++ +/`:

https://github.com/dlang/dlang.org/pull/2133

Reply:
AFAIK, the vast majority of ddocs in Phobos use `/** */`. Changing that to `/++ +/` is going to introduce huge amounts of churn converting from the former to the latter.

Plus, I don't think `/++ +/` gives us very much advantage at all... the only time you'd want comments inside a ddoc is when there are code examples, and Phobos nowadays almost exclusively uses ddoc'd unittests where any comments are actual comments rather than comments embedded inside ddoc comments. Not to mention that comments in code examples tend to be brief (otherwise they really belong in the ddoc text rather than inside the example), so `//` would serve just as well.

Reply:
Both styles are used heavily in Phobos, so if we want to normalize the docs, we'd be changing a ton of code either way. I'd be fine with not requiring one over the other, but I see zero advantage to `/** */` if we're going to require one or the other.

Reply:
both are good stars are preferred

Reply:
If both are considered okay, then we shouldn't require one or the other in the style guide but should just state that we don't want leading stars or pluses on each line.

On a related note, I don't know why we'd want to not indent the documentation. I believe that most of the existing ddoc comments _are_ indented, though fortunately, most don't use the leading stars.

Reply:
https://github.com/dlang/dlang.org/pull/2134